### PR TITLE
Issue 3466 - Unhandled error in edit profile shows up in the non-open tabs

### LINKED ISSUE #3466: Unhandled error in edit profile shows up in the non-open tabs
## Description
When the edit profile is open, and an error is triggered in one of the tabs, the other tabs raise the unexpected error.

- This is happening in: staging
- This requires a hotfix to production: no

### Steps to replicate
- Open the edit profile modal;
- Go to the password tab;
- Try to update the password using an incorrect `old password`
- Go to integrations or personal tab

### Current Behaviour
- Unexpected error shows up

### Expected Behaviour
- Unexpected error should not show up

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #3466 

#### Description
Tabs are now unmounted and re-mounted on tab switch (previously they were just being hidden);
To achieve this, most of the logic of the single tabs was moved in the parent component (the `EditProfileModal`) and the `FormProvider` from react-hook-form was used to have the single tabs form-data persist on tab switch

The `UnhandledError` component now just takes in an error and the array of expected error validators, whereas the parent component that renders it, has access to a new hook called `useErrorInterceptor` that provides access to errors raised by an API call

### CODE REVIEW

**Comment by The-Daniel** [comment]
image
Change password no longer works. The submit button isn't enabled

**Comment by The-Daniel** [comment]
There's a weird double scroll effect when you successfully update your profile and the success message appears. Also the scrollbar takes up the entire height of the modal which isn't useful, but hopefully fixing the other problem will fix this

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

**Comment by Amantini1997** [comment]
> There's a weird double scroll effect when you successfully update your profile and the success message appears. Also the scrollbar takes up the entire height of the modal which isn't useful, but hopefully fixing the other problem will fix this

To separate that concern form the rest of the logic of this PR, this will be fixed as part of this issue https://github.com/3drepo/3drepo.io/issues/3570

File: frontend/src/v5/ui/routes/dashboard/projects/projectsList/createProjectModal/createProjectModal.component.tsx [hunk 1]
```diff
@@ -26,7 +26,9 @@ import { FormSelect } from '@controls/formSelect/formSelect.component';
 import { MenuItem } from '@mui/material';
 import { TeamspacesHooksSelectors } from '@/v5/services/selectorsHooks/teamspacesSelectors.hooks';
 import { projectAlreadyExists } from '@/v5/validation/errors.helpers';
-import { UnhandledError } from '@controls/errorMessage/unhandledError/unhandledError.component';
+import {
+	UnhandledErrorInterceptor,
+} from '@controls/errorMessage/unhandledErrorInterceptor/unhandledErrorInterceptor.component';
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/projects/projectsList/createProjectModal/createProjectModal.component.tsx:31]:
```suggestion
import { UnhandledErrorInterceptor } from '@controls/errorMessage/unhandledErrorInterceptor/unhandledErrorInterceptor.component';
```

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Review by The-Daniel — APPROVED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Pull request merged**